pkg/distributor/distributor.go

```diff
@@ -1128,15 +1126,17 @@ func (d *Distributor) checkWriteRequestSize(rs *requestState, writeRequestSize i
 
 	rs.writeRequestSize = writeRequestSize
 	inflightBytes := d.inflightPushRequestsBytes.Add(writeRequestSize)
-	return d.checkInflightBytes(inflightBytes)
+	return d.checkInflightBytes(inflightBytes, false)
 }
 
-func (d *Distributor) checkInflightBytes(inflightBytes int64) error {
+func (d *Distributor) checkInflightBytes(inflightBytes int64, rejectEqualInflightBytes bool) error {
```

Comment:
The `rejectEqualInflightBytes` looks a complication with a minimal benefit. I don't think we need a limit which is precise at the byte, but just good enough. Can we just remove it?

Reply:
Fair enough, I've simplified it in distributor.

Ingester has similar check, and I quickly tried to remove it too, but then tests got more complicated, so I left it as-is.